File: tests/machine/x/zig/in_operator_extended.zig
const std = @import("std");

fn handleError(err: anyerror) noreturn {
    std.debug.panic("{any}", .{err});
}

fn _contains_list_int(v: []const i32, item: i32) bool {
    for (v) |it| { if (it == item) return true; }
    return false;
}

const xs = &[_]i32{
    1,
    2,
    3,
}; // []const i32
const ys = blk0: { var _tmp0 = std.ArrayList(i32).init(std.heap.page_allocator); for (xs) |x| { if (!((@mod(x, 2) == 1))) continue; _tmp0.append(x) catch |err| handleError(err); } const _tmp1 = _tmp0.toOwnedSlice() catch |err| handleError(err); break :blk0 _tmp1; }; // []const i32
const M = struct { a: i32, };
const m = M{ .a = 1 }; // M
const s = "hello"; // []const u8

pub fn main() void {
    std.debug.print("{}\n", .{_contains_list_int(ys, 1)});
    std.debug.print("{}\n", .{_contains_list_int(ys, 2)});
    std.debug.print("{any}\n", .{_contains_list_int(m, "a")});
    std.debug.print("{any}\n", .{_contains_list_int(m, "b")});
    std.debug.print("{}\n", .{_contains_list_int(s, "ell")});
    std.debug.print("{}\n", .{_contains_list_int(s, "foo")});
}

File: tests/machine/x/lua/in_operator_extended.lua
function __contains(container, item)
    if type(container) == 'table' then
        if container[1] ~= nil or #container > 0 then
            for _, v in ipairs(container) do
                if v == item then return true end
            end
            return false
        else
            return container[item] ~= nil
        end
    elseif type(container) == 'string' then
        return string.find(container, item, 1, true) ~= nil
    else
        return false
    end
end
xs = {1, 2, 3}
ys = (function()
    local _res = {}
    for _, x in ipairs(xs) do
        if ((x % 2) == 1) then
            _res[#_res+1] = x
        end
    end
    return _res
end)()
print(__contains(ys, 1))
print(__contains(ys, 2))
m = {["a"]=1}
print((m["a"] ~= nil))
print((m["b"] ~= nil))
s = "hello"
print(__contains(s, "ell"))
print(__contains(s, "foo"))
